### DISCUSSION TITLE
Excuse me，how to customize error messages when validating forms?

### DISCUSSION THREAD

**Discussion by wfsdaj** [leafsphp/leaf#118] [Q&A] [answered]
The default error message is in English. How to better customize other languages?

**Comment by mychidarko** [comment, marked as answer]
Currently, the error messages displayed are the defaults set in leaf form. This version doesn't allow you to customize these error messages, however, from the next version, you'll be able to do so.

**Reply by mychidarko** [reply]
@wfsdaj this is now possible with the latest version of leaf form. You can now customize messages like this:

```php
form()->messages('nospaces', '{field} 不能包含空格');
```

The current field will replace `{field}`. You can view all the details with the link below.

https://github.com/leafsphp/form/releases/tag/v1.2

**Reply by wfsdaj** [reply]
```
form()->messages('min', '{field} 不能少于6个字符。');

$validation = form()->validate([
    "username" => ["required", "username", "min:6"],
    "email"    => ["required", "email"],
    "password" => ["required", "min:8"],
]);
```

If there are the same validation rules, the error message will be repeated.

**Reply by wfsdaj** [reply]
Some frameworks do this:

```
form()->messages([
    'username.min' => '{field} 不能少于6个字符。',
    'password.min' => ''{field} 不能少于8个字符。',
]);

$validation = form()->validate([
    "username" => ["required", "username", "min:6"],
    "email"    => ["required", "email"],
    "password" => ["required", "min:8"],
]);
```
